lib/ts/utils.ts

```diff
@@ -556,6 +559,121 @@ export function useRethrowInRender() {
     return setError;
 }
 
+export async function jwtVerify<T extends { exp?: number; nbf?: number; iat?: number }>(
+    token: string,
+    jwksUrl: string
+): Promise<T> {
+    const parts = token.split(".");
+    if (parts.length !== 3) {
+        throw new Error("Invalid JWT token format");
+    }
+
+    const [headerB64, payloadB64, signatureB64] = parts;
+
+    const headerStr = Buffer.from(headerB64 + "=".repeat((4 - (headerB64.length % 4)) % 4), "base64").toString();
+    const header = JSON.parse(headerStr);
+
+    const kid = header.kid;
+    if (!kid) {
+        throw new Error("JWT header missing kid (Key ID)");
+    }
+
+    const alg = header.alg as string;
+    if (!alg) {
+        throw new Error("JWT header missing alg (Algorithm)");
+    }
+
+    const jwksResponse = await fetch(jwksUrl);
```

Comment:
This will reload the jwks on every action call, could we cache this somehow? Also, I'm not necessarily opposed to using jose if it doesn't mess with our bundle sizes.